### ISSUE TITLE
Inconsistent Controllability Results on ML-1M Compared to Paper

### ISSUE THREAD

**Issue by YTian999** [Emilianopp/TEARS#3]
Hi,

Thanks for your great work on TEARS and for making the code publicly available!

I've been trying to reproduce the results for the **Large Scope Changes** experiment (controllability) on the **ML-1M** dataset. However, the results I'm getting seem quite different from what's reported in the paper.

I ran the evaluation script and got the following `Δ_up` (`NDCG up`) and `Δ_down` (`NDCG down`) values for the various models:

|Model|My `Δ_up` Result|My `Δ_down` Result|
|---|---|---|
|`TearsBase`|-0.072|0.130|
|`TearsMVAE`|-0.089|0.139|
|`TearsMacrid`|-0.004|-0.039|
|`TearsRecVAE`|-0.090|0.118|

When I compare this to Table 7 and Figure 4 in the paper for the ML-1M dataset (using GPT-generated summaries at `α=1`), the reported values are significantly different and consistently positive. For example, the paper reports:

|Model|Paper's `Δ_up` (Table 7)|Paper's `Δ_down` (Table 7)|
|---|---|---|
|`TearsMulti-VAE`|0.201|0.193|
|`TearsMacridVAE`|0.690|0.453|
|`TearsRecVAE`|0.391|0.346|

As you can see, my results for `Δ_up` are negative, and the magnitudes for both metrics are much smaller than those in the paper.

I was wondering if this is expected behavior, or if perhaps I'm missing a specific configuration or setup step? Any clarification would be greatly appreciated.

Here is my reproduction repository: https://github.com/fudan-dcy/TEARS/tree/master

Thanks for your help!

Running Log: 
---

TearsBase loaded

Some weights of TearsVAE were not initialized from the model checkpoint at google-t5/t5-base and are newly initialized: ['mlp.0.weight']

You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.

TearsMVAE loaded

Some weights of MacridTEARS were not initialized from the model checkpoint at google-t5/t5-base and are newly initialized: ['mlp.0.weight']

You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.

TearsMacrid loaded

Some weights of TearsVAE were not initialized from the model checkpoint at google-t5/t5-base and are newly initialized: ['mlp.0.weight']

You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.

TearsRecVAE loaded

Evaluating model: TearsBase

Controllability:   0%|                                                                                                                                | 0/1 [00:00<?, ?it/s]Asking to truncate to max_length but no maximum length is provided and the model has no predefined maximum length. Default to no truncation.

Average NDCG up: -0.07159449388364407 | Average NDCG down: 0.12992506363436698                                                                                              

Controllability: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:02<00:00,  2.82s/it]

Final average NDCG up: -0.07159449388364407 | Final average NDCG down: 0.12992506363436698

Evaluating model: TearsMVAE

Average NDCG up: -0.08928266746781505 | Average NDCG down: 0.1388599464065452                                                                                               

Controllability: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:02<00:00,  2.65s/it]

Final average NDCG up: -0.08928266746781505 | Final average NDCG down: 0.1388599464065452

Evaluating model: TearsMacrid

Average NDCG up: -0.003752475581025755 | Average NDCG down: -0.03922174970481805                                                                                            

Controllability: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:02<00:00,  2.71s/it]

Final average NDCG up: -0.003752475581025755 | Final average NDCG down: -0.03922174970481805

Evaluating model: TearsRecVAE

Average NDCG up: -0.09025267753786224 | Average NDCG down: 0.11789815004161389                                                                                           

Controllability: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:02<00:00,  2.67s/it]

Final average NDCG up: -0.09025267753786224 | Final average NDCG down: 0.11789815004161389

**Comment by Emilianopp** [comment]
Hello @fudan-dcy!

Thanks for taking interest in the work and running our code a few points:

The negative values are intendeed behaviour for NDCG up (we take the absolute value in the paper).

You are right the scores do seem low, do you happen to know what \alpha value you are using?

Also are you training things from scratch yourself? Or are these using the checkpoints provided, are the summaries being made by yourself or are you using the provided ones?  This could all have a relatively large impact on performance. 

Please let me know :)

**Comment by YTian999** [comment]
Hi,

Thanks so much for the quick and helpful response\!

To answer your questions:

  * **Regarding the `α` value:** You're right to ask about it. I was using `α=0.5` in my `run_control.sh` script.
  * **Training and Summaries:** I trained all the models from scratch myself. I did, however, use the provided user summaries from `saved_user_summary.tar.gz`.

You mentioned provided checkpoints, which would be really useful for debugging. However, I couldn't locate them in the repo. I only found `data.tar.gz`, `data_preprocessed.tar.gz`, and `saved_user_summary.tar.gz`. Could you please let me know where I can find the pre-trained checkpoints?

**Comment by Emilianopp** [comment]
Hey @fudan-dcy ,

I see I think I may have had a mistake in the number of epochs that TEARS models should be ran for. 

We ran them for ~200 epochs. I found this makes a considerable difference (specially in the controllability). 

If you can run them for that long and let me know I would appreciate it. 

If this still does not resolve the issue I'll try to get you some checkpoints on my end and make a new pull request with the required changes.

In the paper's appendix we have some more details on the hyper-parameters I would highly suggest referencing this too!

**Comment by YTian999** [comment]
@Emilianopp,
Thank you for the clarification! I’ll try running the models with the updated settings and let you know the results. I’ll also check the appendix for further hyper-parameter details. Appreciate your help!

**Comment by YTian999** [comment]
Hi @Emilianopp , thank you for your great work on this project. I have question after running experiments for 200 epochs.

I've noticed that the controllability metrics (`Average NDCG up/down`) for TearsMVAE, TearsMacrid, and TearsRecVAE are lower than those reported in the paper for the ML-1M dataset at both `alpha=0.5` and `alpha=1`.

- alpha = 1

> === Evaluating TearsBase ===
> Average NDCG up: -0.4193133457144453 | Average NDCG down: 0.30610239524999605                                                                                                     
> 
> === Evaluating TearsMVAE ===
> Average NDCG up: -0.19281035714886238 | Average NDCG down: 0.24447321877882613                                                                                                    
> 
> === Evaluating TearsMacrid ===
> Average NDCG up: -0.5675509941035206 | Average NDCG down: 0.4408333425591776                                                                                                      
> 
> === Evaluating TearsRecVAE ===
> Average NDCG up: -0.24037425549248895 | Average NDCG down: 0.2657411112859934

**Comment by Emilianopp** [comment]
Hey @YTian999 

Thanks for taking the time to reproduce the results they look more consistent now! 

In all honesty the discrepancies are minor. I did a lot of tuning for both the base VAE and the TEARS models, and both of these could contribute to this. 

The discrepancies could also be due to seed. 

If you would like to optimize the model more I would mostly just suggest sweeping over more hyper-parameters, but your printed results look fairly decent to me :).

**Comment by YTian999** [comment]
Thanks for your clarifying, this resolves my concerns!
